### PR TITLE
refined code by components

### PR DESCRIPTION
**Pull request by sardhara-preet**

### CODE REVIEW

File: api/slack/receive-message/receive-message.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
-const request = require('request');
-const handlers = require("./handlers");
+const Slack = require("../../../components/slack");
 
 module.exports = async function(req,res) {
     res.status(200).end();
-    const data = await handlers[req.body.command](req.body);
-    request.post(`${process.env.LOCALHOST_URL}/slack/sendmessage`).form(data);
+    const data = await Slack.handleCommand[req.body.command](req.body.text, req.body.user_name);
```
Comment (dvijeshpatel) [inline api/slack/receive-message/receive-message.js]: this should only method not map pass req to this

File: components/slack/handle-commands/show-status.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const SLACK_ENTITY = require("../../../api/slack/slack-entity");
+const Jira = require("../../jira");
+
+module.exports = async function(key, userName) {
+    const issue = await Jira.fetch.fetchIssue(key);
+    if(issue) {
+        const data = {
```
Comment (dvijeshpatel) [inline components/slack/handle-commands/show-status.js]: remove duplicate

File: components/slack/handle-commands/possible-workflow-status.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+function createMessage(transitions) {
+    const message = `Type */changestatus* followed by *Key:ID* of the status you want to change to.\nYou can find the ID from the list given below:\n>${getTransitionList(transitions)}`;
+    return message;
+}
+
+module.exports = async function(key, userName) {
+    const status = await Jira.fetch.fetchTransitions(key);
+    if(!status) {
+        const data = {
```
Comment (dvijeshpatel) [inline components/slack/handle-commands/possible-workflow-status.js]: remove duplicate

File: components/slack/handle-commands/possible-workflow-status.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const SLACK_ENTITY = require("../../../api/slack/slack-entity");
+const Jira = require("../../jira");
+
+function getTransitionList(transitions) {
+    let message = "";
+    transitions.forEach(transition => message+='```• ' + transition + '``` ');
```
Comment (dvijeshpatel) [inline components/slack/handle-commands/possible-workflow-status.js]: use reduce or map

File: components/slack/handle-commands/change-status.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+const SLACK_ENTITY = require("../../../api/slack/slack-entity");
+const Jira = require("../../jira");
+
+module.exports = async function(text, userName) {
+    const pair = text.split(':');
+    const key = pair[0];
```
Comment (dvijeshpatel) [inline components/slack/handle-commands/change-status.js]: use destructuring

File: components/jira/index.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const fetch = require("./fetch");
+const updateStatus = require("./update-status");
+
+const Jira = {
+    fetch: fetch,
```
Comment (dvijeshpatel) [inline components/jira/index.js]: change name fetch to service

File: components/gitlab/payload-creators/message-creator/push/push-message-creator.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ function getCommitMessage(req) {
 }
 
 module.exports = async function(req) {
-    const mergeRequest = await getMergeRequestOfCommit(req.project.id, req.checkout_sha);
+    const mergeRequest = await fetchers.fetchMergeRequest(req.project.id, req.checkout_sha);
```
Comment (dvijeshpatel) [inline components/gitlab/payload-creators/message-creator/push/push-message-creator.js]: change name fetchers to service

File: components/gitlab/payload-creators/message-creator/merge-request/simplify-action.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+const MERGE_REQUEST_ACTIONS = require("./merge-request-actions");
+
+const SIMPLIFIED_ACTIONS = {
```
Comment (dvijeshpatel) [inline components/gitlab/payload-creators/message-creator/merge-request/simplify-action.js]: only ACTIONS

File: components/gitlab/payload-creators/merge-request-payload-creator.js [hunk 9]
```diff
@@ -1,8 +1,7 @@
-const SLACK_ENTITY = require("../../slack/slack-entity");
-const findEmailByID = require("../../gitlab/services/email-fetcher-by-id");
-const notesFetcher = require("../../gitlab/services/notes-fethcher-on-merge-request");
+const SLACK_ENTITY = require("../../../api/slack/slack-entity");
+const fetchers = require("../fetchers");
```
Comment (dvijeshpatel) [inline components/gitlab/payload-creators/merge-request-payload-creator.js]: change name to Service

File: components/slack/handle-commands/change-status.js (hunk 5 above)
Comment (sardhara-preet) [reply components/slack/handle-commands/change-status.js]: const [key, id] = pair

File: app.js [hunk 10]
```diff
@@ -1,14 +1,44 @@
 const express = require('express');
-require('dotenv').config()
-const slack = require("./api/slack")
+const request = require('request');
+require('dotenv').config();
+const slack = require("./api/slack");
```
Comment (dvijeshpatel) [inline app.js]: convert all double quote to single

File: app.js [hunk 11]
```diff
@@ -1,14 +1,44 @@
...
 const gitlab = require("./api/gitlab");
-const jira = require("./api/jira");
+const Jira = require("./components/jira");
 
-const app = express();
+function subscribeToWebhook() {
+    const body = {
```
Comment (dvijeshpatel) [inline app.js]: move constant to outside,  name subscribeToGitlabWebhook

File: app.js [hunk 12]
```diff
@@ -1,14 +1,44 @@
...
+        id: 12918952,
```
Comment (dvijeshpatel) [inline app.js]: constant for id

File: components/gitlab/update-issue.js/merge-request/in-dev-review.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+        type: SLACK_ENTITY.USER,
+        email: user.emailAddress,
+        message: `The status of *Issue: ${key}* has automatically been changed to *${status}* due to Creation of its Merge Request.`
+    }
+    return data;
+}
+
+module.exports = async function(req) {
+    const branch = req.object_attributes.source_branch;
+    const key = branch.slice(branch.lastIndexOf('/')+1).toUpperCase();
```
Comment (dvijeshpatel) [inline components/gitlab/update-issue.js/merge-request/in-dev-review.js]: if key not found make function for finding key

**Pull request merged**